vars/dockerBuildArgs.groovy

```diff
@@ -49,26 +101,43 @@ String call(Map config = [:]) {
       ret_str += " --build-arg CB0=" + current_time.get(Calendar.WEEK_OF_YEAR)
     }
 
-    if (add_repos) {
-      if (env.REPOSITORY_URL) {
-        ret_str += ' --build-arg REPO_URL=' + env.REPOSITORY_URL
-      }
-      if (env.DAOS_STACK_EL_7_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL7=' + env.DAOS_STACK_EL_7_LOCAL_REPO
+    String repo_base = env.REPOSITORY_URL
+    // No repo_base, no repos to add.
+    if (add_repos && repo_base) {
+      String repo_name = null
+      String repo_arg = ''
+      if (stage_info['target'] == 'centos7') {
+        repo_alias = 'EL_7'
+        repo_arg = 'EL7'
       }
-      if (env.DAOS_STACK_EL_8_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL8=' + env.DAOS_STACK_EL_8_LOCAL_REPO
+      if (stage_info['target'] == 'centos8') {
+        repo_alias = 'EL_8'
+        repo_arg = 'EL8'
       }
-      if (env.DAOS_STACK_LEAP_15_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_LOCAL_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_LOCAL_REPO
+      if (stage_info['target'] == 'leap15') {
+        repo_alias = 'LEAP_15'
+        repo_arg = 'LEAP'
+        // Backwards compatibilty for LOCAL
+        if (repo_type == 'LOCAL') {
+         if (env.DAOS_STACK_LEAP_15_GROUP_REPO) {
+            ret_str += ' --build-arg REPO_GROUP_LEAP15=' +
+                       env.DAOS_STACK_LEAP_15_GROUP_REPO
+          }
+        }
       }
-      if (env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_UBUNTU_20_04=' + env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+      if (stage_info['target'] == 'ubuntu20.04') {
+        // Ubuntu repos curently only used for package building.
+        // When fully implemented it will probably be similar to above.
+        // And the URLS for will be for installing a list of repos.
+        // Details still to be worked out.
+        repo_alias = 'UBUNTU_20_04'
+        repo_arg = repo_alias
       }
-      if (env.DAOS_STACK_LEAP_15_GROUP_REPO) {
-        ret_str += ' --build-arg REPO_GROUP_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_GROUP_REPO
+      repo_name = env["DAOS_STACK_${repo_alias}${repo_mod}_${repo_type}_REPO"]
+      // Only add the build args if a repo was found.
+      if (repo_name) {
+        ret_str += " --build-arg REPO_${repo_arg}=" + repo_name
+        ret_str += ' --build-arg REPO_URL=' + env.REPOSITORY_URL
```

Comment:
Similar to the comment above about assigning `env.*` to local vars, you already have `repo_base` containing this value, so some inconsistency at least.  But as I said before, I find the direct use of `env.REPOSITORY_URL` here more transparent and less burdensome than having to keep track of/remember the local variable (while reading the code), for what I believe is about the same cost.

Reply:
I was thinking that I needed to assign the env.REPOSITORY_URL at the top to a variable to make it simpler to test if it was not null.   That is obviously not needed.

Will make another push.